### ISSUE TITLE
Question regarding  iq_loss implementation

### ISSUE THREAD

**Issue by sangwon-seo** [Div99/IQ-Learn#12]
Hi!
First, thanks for sharing the codebase of this great work. 
I have a question regarding the implementation of `iq_loss` function which resides in the "iq.py" file.

In line 26-43, I don't quite understand why you used the gradient of \phi instead of directly following Equation (9) of the IQLearn Paper with \phi in Table 4. I thought we can represent loss function just the same as the equation because pytorch will automatically calculate gradients via `autograd`. I wonder what I'm missing here.

**Comment by Div-Infinity** [comment]
Hi, yes we used phi_grad initially as were worried about the numerical stability of the gradients in our experiments for different f-divergences, but agreeing with your point, you can use phi and torch `autograd` can handle it well

**Comment by sangwon-seo** [comment]
Got it! Thanks for the explanation.
